### PR TITLE
Add ingress domain

### PR DESCRIPTION
**Pull request by Miciah**
* `config/v1/types_ingress.go` (`IngressSpec`): Add `Domain` field.

Related to [NE-100](https://jira.coreos.com/browse/NE-100).

### CODE REVIEW

File: config/v1/types_ingress.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ type Ingress struct {
 }
 
 type IngressSpec struct {
-	// default suffix.  It goes here or it gets removed from server
+	// Domain specifies the default subdomain for routes.
```
Comment (deads2k) [inline config/v1/types_ingress.go]: lower "d" in API docs to match the json string.

**Comment by deads2k** [comment]
@ironcladlou confirm that you expect and want the domain to be user changeable.  That's a significant change.

File: config/v1/types_ingress.go (hunk 1 above)
Comment (abhinavdahiya) [reply config/v1/types_ingress.go]: Also is it common knowledge for OpenShift users that domain for ingress means subdomain?

**Comment by Miciah** [comment]
According to https://github.com/openshift/api/pull/124#issue-229437940
> Controller/Operator maintained information lives in status, cluster-admin maintained information lives in spec.

I didn't see anything about using spec for mutable fields and status for immutable fields.  Is that documented?

**Comment by deads2k** [comment]
> According to [#124 (comment)](https://github.com/openshift/api/pull/124#issue-229437940)
> 
> > Controller/Operator maintained information lives in status, cluster-admin maintained information lives in spec.
> 
> I didn't see anything about using spec for mutable fields and status for immutable fields. Is that documented?

It's about user specifiable or not.  The kube convention is that user specifiable fields are in spec, non-user specifiable ones are in status.

**Comment by Miciah** [comment]
It's user-specifiable, but only once (during installation), and it's not anyone-changeable (neither users nor operators should be changing it).  Documentation could be clearer.  I see your point though and will update the PR shortly.

**Comment by deads2k** [comment]
/approve

@ironcladlou your API.  If you're happy with the field name and doc, you can lgtm.  Think of this as your doc that will explain to an admin what it does and to developers how to use it.

/assign @ironcladlou

File: config/v1/types_ingress.go [hunk 2]
```diff
@@ -20,10 +20,12 @@ type Ingress struct {
 }
 
 type IngressSpec struct {
-	// default suffix.  It goes here or it gets removed from server
 }
 
 type IngressStatus struct {
+	// domain specifies the default subdomain that will be used to compose
```
Comment (knobunc) [inline config/v1/types_ingress.go]: How about:

domain is used to generate the hostname for a route when none is specified.  The pattern that is used is: routename-projectname.domain.

**Comment by knobunc** [comment]
@deads2k it feels really weird for something to read its state from status not spec.  I would argue that it is mutable, but doing so ought to be a rare event.  So I'd lean for spec.  Thoughts?

**Comment by ironcladlou** [comment]
I was wrong, it's mutable after all. Back to spec it goes!

**Comment by Miciah** [comment]
And with the latest push, `Domain` has gone back to `IngressSpec`.  Now with more docstring!

**Pull request merged**